core/lib/spree/testing_support/preferences.rb

```diff
@@ -75,22 +84,29 @@ def stub_spree_preferences(preferences)
       #     expect(Spree::Config.currency).to eql 'EUR'
       #   end
       # @see Spree::TestingSupport::Preferences#stub_spree_preferences
-      def with_unfrozen_spree_preference_store
-        frozen_store = Spree::Config.preference_store
-        Spree::Config.preference_store = Spree::Config[:unfrozen_preference_store].dup
+      def with_unfrozen_spree_preference_store(preference_store_class: Spree::Config)
+        frozen_store = preference_store_class.preference_store
+        preference_store_class.preference_store = preference_store_class[:unfrozen_preference_store].dup
         yield
       ensure
-        Spree::Config.preference_store = frozen_store
+        preference_store_class.preference_store = frozen_store
+      end
+
+      def self.freeze_preferences(config_class:, preference_store_class:)
```

Comment:
I'd like not to define a module-level method here, but I still can't find a better way to do this since I need to use this method into the `conf.before` block. Any idea?

Reply:
I'm OK with this. It makes the idea of name spacing configs per extension easy to use and understand.